File: solutions/python/largest-series-product/1/largest_series_product.py
def largest_product(series, size):
    if size > len(series):
        raise ValueError("span must be smaller than string length")
    if any(not c.isdigit() for c in series):
        raise ValueError("digits input must only contain digits")
    if size < 0:
        raise ValueError("span must not be negative")
    
    max_product = 0

    for i in range(len(series) - size + 1):
        curr = 1
        for n in range(i, i + size):
            curr *= int(series[n])
        max_product = max(curr, max_product)
    
    return max_product

File: solutions/go/largest-series-product/1/largest_series_product.go
package lsproduct

import "errors"

// LargestSeriesProduct returns the largest product of length span from digits.
func LargestSeriesProduct(digits string, span int) (int64, error) {
	if len(digits) < span {
		return 0, errors.New("span must be smaller than string length")
	} else if span < 0 {
		return 0, errors.New("span cannot be negative")
	}

	var max int64 = 0

	for i := 0; i < len(digits) - span + 1; i++ {
		if digits[i] < '0' || digits[i] > '9' {
			return 0, errors.New("digits input must only contain digits")
		}

		curProd := int64(digits[i] - '0')

		for j := 1; j < span; j++ {
			curProd *= int64(digits[i + j] - '0')
		}

		if curProd > max {
			max = curProd
		}
	}

	return max, nil
}
